internal/containerd/client.go

```diff
@@ -0,0 +1,18 @@
+package containerd
+
+import (
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/defaults"
+)
+
+var (
+	defaultAddr = defaults.DefaultAddress
```

Comment:
does this address change with a rootless configuration? (say somewhere under `/run/user/$(id -u)/containerd...`)

Reply:
No it does not. It always tries to use `/var/run/containerd/containerd.sock`. However, when I manually updated the file to point to the `api.sock` for rootless it still worked fine so I just to find a way to wire that in.

Reply:
I take this back actually, seems it may not work with that. From what I know of containerd's command line tool `nerdctl` they also use that same `defaults.DefaultAddress`. Inside of the docs, [rootless containerd](https://github.com/containerd/containerd/blob/main/docs/rootless.md), it doesn't specify a different address in the `nerdctl` command so it would use that default value still. So I assume for root and rootless this works maybe?

Reply:
I think I have something working for this. I plan to make a PR to your syft branch for adding the containerd address flag. The way I am currently passing the value from `syft` to `stereoscope` is through setting a value in the context, would it be preferred that I send it via a parameter instead? In `syft` I had to add `containerdAddress string` to bunch of methods. The only way I could find getting around this was adding it as an `Application` attribute. However, when saving to a file then you would be get this value mapped to your file, which if we want is fine but I figured we wouldn't that in an output file.

Reply:
I pushed up a change to that should fix the integration tests. Also allows the passing of the `containerdAddress` to different functions so the client which use that address if specified otherwise will try to use the default address. There will be a matching PR for `syft`